src/CMakeVersion.txt

```diff
@@ -0,0 +1,14 @@
+set(GIT_CMD "git")
+set(GIT_ARGS "log" "-n" "1" "--pretty=format:%h - %ad")
+execute_process(COMMAND ${GIT_CMD} ${GIT_ARGS}
+    RESULT_VARIABLE GIT_RESULT
+    OUTPUT_VARIABLE GIT_COMMIT)
+    if (GIT_RESULT)
+        set (GIT_COMMIT "UNKNOWN")
+    endif()
+set(ZMAP_VERSION "Development Build. Commit ${GIT_COMMIT}")
+
+configure_file(topt.ggo.in topt.ggo @ONLY)
```

Comment:
1. Point taken.
2. I wanted to make sure the dependencies were still picked up correctly, and that this was triggered after a modification.
3. Seemed easier to implement in an easier language, especially being able to explicitly note dependency chains in `add_custom_command`. See (2)